Marlin/src/module/planner.cpp

```diff
@@ -1395,7 +1392,12 @@ void Planner::check_axes_activity() {
   // Update Fan speeds
   // Only if synchronous M106/M107 is disabled
   //
-  TERN_(HAS_TAIL_FAN_SPEED, if (fans_need_update) sync_fan_speeds(tail_fan_speed));
+  #if HAS_TAIL_FAN_SPEED
+    if (fans_need_update) {
+      sync_fan_speeds(tail_fan_speed);
+      fans_need_update = false;
```

Comment:
`fans_need_update` is not static, but is recalculated every time. So this does nothing.